### ISSUE TITLE
Request: Item Descriptions & Repair Rolls

### ISSUE THREAD

**Issue by AtomicBrainBag** [free-beer/bh2e#24]
Hello again,

My party and I have now fully transitioned over from 5e to BH2e. We loved our first session, however, there were a few sticking points.

1. Item descriptions - We would love to be able to either click on or mouse over and item/weapon in the inventory and have the information become visible. Many of the items we create have effects and abilities, and remembering them all is a challenge. Is this something that could be added?

2. Repair rolls - currently, repair rolls need to be done manually. This is a bit of an inconvenience, especially as usage dice rolls are done automatically. Is there any way that, when a player clicks the repair button, the roll-over mechanic for armour is done in the background and the result is pushed to the chat?

Thank you so much for making this system, my party and I love it so far and would love to see these implemented to make this infinitely more smooth for everyone's play.

Thanks a million, 
Andy

**Comment by free-beer** [comment]
The first of these is relatively straight forward but will take a fair bit of work. The second one I'm not clean on, can you provide a more detailed description of what it is that is causing the issue?

**Comment by AtomicBrainBag** [comment]
Hey,

Thanks for looking at this. I would be more than happy to elaborate.

From what I understand, the armour repair rolls are currently just decremented when you click the lightning bolt icon, tool tipped as "Break Die" button. This is fine as it reduces the number of total dice and makes sense to be manually done, as it is player choice.

The issue is, when you have a rest and repair the broken armour dice (D6), you roll the armour dice (D6) to do so. A value one or under the armours AV, the dice is broken permanently until it is repaired. If all the armour dice are broken, the armour is destroyed. For reference, **this is on p.g. 10 of the rule book under Armour.**

Currently, the system, as I can tell, only increments the armour value back up. Would it be possible to add functionality that allows this to be done as per the black hack system.

I hope this clears things up for you, any more questions, please ask away.

Andy

**Comment by free-beer** [comment]
Okay, having thought about this a little more the armour thing adds a level of complication I'm not happy with. There are two primary aspects of concern...

1. Having read back through the armour rules they indicate that, when rolling to repair armour dice after a rest, the total you're aiming for is the total on your main suit of armour. Helmets and shields _"add +1 each to the pool size, they do not modify the armours **AV** value in any way._". That means I'd have to be able to pick out what your 'suit' of armour is from all of your armour items and there isn't currently a mechanism for doing this. Though I could introduce one I'm not sure that's a good idea as it would mean filtering helmets and shields out as separate entities in some way.
2. The rules also state that a die that fails the armour repair roll is "_**Broken** permanently_". This introduces a new state for an armour die that I'd have to record and take into account in the functionality.

Looking at these two points I think these are fundamentally the reason why I implemented the armour die system the way it is. What's there is simple and gives full control. I'll think about it a little more but my feeling at the moment is that it's not worth introducing this complication.

**Comment by AtomicBrainBag** [comment]
Hey, 

Thanks for getting back to me about it. Really happy you gave it some thought. Totally understandable. Running through it like that does sound like a bunch of work that would possibly break things as they are.

On the first idea, though, the item descriptions. Would that be something you would be comfortable working on? No pressure if not.

Thanks a bunch, 
Andy

**Comment by free-beer** [comment]
Yeah, the item descriptions is a fine request and I've already done that on other systems I've created. I've already ported that work over and I'm just waiting to see if I can think of something else to add in before going for a release with it. Just to clarify, the descriptions you are looking for is primarily on equipment items (i.e. weapons, armour and general equipment)?

**Comment by AtomicBrainBag** [comment]
Hey, that's spot on! That way we can look at the abilities each weapon may have easier than opening up the compendium etc.

Cheers for all of this :)

**Comment by apewall** [comment]
As an alternative;
Please consider adding a simple tracker for armour die to the character sheet and allowing the player to keep track themselves of the number of armour die they have available.

It would also be nice if you could track whether a piece of armor is "broken" on the character sheet.

**Comment by widget** [comment]
I'd also be interested in the artwork for the item also being visible with the description if it would fit (this doesn't feel like a separate request, but happy to raise it as one)
